src/basic/compress.c

```diff
@@ -91,6 +91,20 @@ static int zstd_ret_to_errno(size_t ret) {
 }
 #endif
 
+#if HAVE_XZ
+static void *lzma_dl = NULL;
+
+static DLSYM_FUNCTION(lzma_code);
+static DLSYM_FUNCTION(lzma_easy_encoder);
+static DLSYM_FUNCTION(lzma_end);
+static DLSYM_FUNCTION(lzma_stream_buffer_encode);
+static DLSYM_FUNCTION(lzma_stream_decoder);
+
+static inline void sym_lzma_endp(lzma_stream *ls) {
+        sym_lzma_end(ls);
+}
```

Comment:
why not DEFINE_TRIVIAL_CLEANUP_FUNC()?

Reply:
Because `lzma_end()` accepts a pointer to lzma_stream, but all the variables are defined as regular variables, so both of them won't work:

```
DEFINE_TRIVIAL_CLEANUP_FUNC(lzma_stream, sym_lzma_end); // expected ‘lzma_stream *’ but argument is of type ‘lzma_stream’
DEFINE_TRIVIAL_CLEANUP_FUNC(lzma_stream*, sym_lzma_end); // passing argument 1 of ‘sym_lzma_endp’ from incompatible pointer type
```

Reply:
If I'm missing something obvious just let me know what I'm wrong, but I didn't find a way to use that macro.

Reply:
@teknoraver Do we need `sym_lzma_endp` at all? Why doesn't `_cleanup_(sym_lzma_end) lzma_stream s = LZMA_STREAM_INIT;` work?

Reply:
It complains about `sym_lzma_end` not being a function, maybe because it's declared as a function pointer due to the dlopen?
```
../src/basic/compress.c: In function ‘decompress_blob_xz’:
../src/basic/compress.c:304:9: error: cleanup argument not a function
  304 |         _cleanup_(sym_lzma_end) lzma_stream s = LZMA_STREAM_INIT;
      |         ^~~~~~~~~
```

Reply:
The preprocessed source files declares it as:
```c
static typeof(lzma_end)* sym_lzma_end = ((void *)0);
```

Reply:
I've added a comment to explain why this wrapper is needed